### PR TITLE
Enable uefi support for caasp

### PR DESCRIPTION
**Pull request by kravciak**
Extend is_casp to accpet $flavor argument

poo#15874

Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/295

Local run:
http://dhcp91.suse.cz/tests/3717 VMX
http://dhcp91.suse.cz/tests/3718 VMX+UEFI
http://dhcp91.suse.cz/tests/3676 DVD
http://dhcp91.suse.cz/tests/3672 DVD+UEFI

### CODE REVIEW

File: lib/bootloader_setup.pm [hunk 1]
```diff
@@ -88,14 +88,16 @@ sub bootmenu_default_params {
     if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
         type_string " video=hyperv_fb:1024x768";
     }
+    return if is_jeos || is_casp('VMX');
```
Comment (sysrich) [inline lib/bootloader_setup.pm]: Wouldn't it make more sense to have this below line 93? Otherwise you're going to skip the Y2DEBUG, which you probably still want

File: tests/installation/bootloader_uefi.pm [hunk 2]
```diff
@@ -117,10 +117,8 @@ sub run() {
     }
 
     type_string " \\\n";    # changed the line before typing video params
-                            # https://wiki.archlinux.org/index.php/Kernel_Mode_Setting#Forcing_modes_and_EDID
```
Comment (okurz) [inline tests/installation/bootloader_uefi.pm]: unrelated change

File: lib/utils.pm [hunk 3]
```diff
@@ -128,8 +128,22 @@ sub is_jeos() {
...
+        if ($flavor eq 'VMX') {
+            return !check_var('FLAVOR', 'DVD');
+        }
+        else {
+            return check_var('FLAVOR', $flavor);
+        }
+    }
+    else {
+        return 0;
+    }
```
Comment (okurz) [inline lib/utils.pm]: WDYT about

```perl
my ($flavor) = @_;
return 0 unless check_var('DISTRI', 'casp');
return 1 unless $flavor;
return !check_var('FLAVOR', 'DVD') if ($flavor eq 'VMX');
return check_var('FLAVOR', $flavor);
```

File: lib/bootloader_setup.pm (hunk 1 above)
Comment (kravciak) [inline lib/bootloader_setup.pm]: There is no yast on qcow images, they are set up with cloud-init. Y2DEBUG is still set for ISO image (would be is_casp('DVD'))

File: tests/installation/bootloader_uefi.pm (hunk 2 above)
Comment (kravciak) [inline tests/installation/bootloader_uefi.pm]: I removed it as it's documented inside bootmenu_default_params - this is just unrelated duplicity here - I can open separate PR if needed

File: lib/bootloader_setup.pm (hunk 1 above)
Comment (kravciak) [inline lib/bootloader_setup.pm]: But I might be wrong, does Y2DEBUG it make sense for JeOS?

Comment (sysrich) [inline lib/bootloader_setup.pm]: Y2debug isn't only about producing debug info for the installer

And I imagine we need to test yast inside jeos, maybe even in some small scale for yast also (I am surprised if there is absolutely NO yast in Caasp and that it will stay that way)

Comment (kravciak) [inline lib/bootloader_setup.pm]: No yast for now, not sure about plans. I changed PR to keep Y2DEBUG for now.

**Comment by okurz** [comment]
typo in commit message: accpet

File: lib/bootloader_setup.pm [hunk 4]
```diff
@@ -89,13 +89,16 @@ sub bootmenu_default_params {
         type_string " video=hyperv_fb:1024x768";
     }
     type_string_slow "Y2DEBUG=1 ";
-    if (!is_jeos && (check_var('ARCH', 'i586') || check_var('ARCH', 'x86_64'))) {
+
+    return if is_jeos || is_casp('VMX');
```
Comment (okurz) [inline lib/bootloader_setup.pm]: please add a line of comment above here why it does not make sense to continue for jeos or casp.

File: tests/installation/bootloader_uefi.pm [hunk 5]
```diff
@@ -117,10 +117,8 @@ sub run() {
...
     bootmenu_default_params;
     specific_bootmenu_params;
-
```
Comment (okurz) [inline tests/installation/bootloader_uefi.pm]: also unrelated change. If you don't change it I will merge the PR nevertheless after fixing the typo in the commit message but in general we should keep this in mind so that the history of changes looks clean and we are not changing whitespace back and forth because of personal taste.

File: tests/installation/bootloader_uefi.pm (hunk 2 above)
Comment (okurz) [inline tests/installation/bootloader_uefi.pm]: separate commit in same PR should be fine

**Comment by kravciak** [comment]
Fixed

**Comment by okurz** [comment]
sorry, forgot to send the review earlier

**Comment by kravciak** [comment]
Fixed
